--- libzip-0.11.2/libzip.pc.in.0000~	2012-03-20 13:54:38.000000000 +0100
+++ libzip-0.11.2/libzip.pc.in	2014-06-28 22:43:17.081354449 +0200
@@ -2,7 +2,6 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-libincludedir=@libdir@/@PACKAGE@/include
 
 zipcmp=@prefix@/bin/zipcmp
 
@@ -10,5 +9,5 @@ Name: libzip
 Description: library for handling zip archives
 Version: @VERSION@
 Libs: -L${libdir} -lzip @LIBS@
-Cflags: -I${includedir} -I${libincludedir}
+Cflags: -I${includedir}
 
