--- libzip-0.11.2/lib/zip_close.c.0002~	2014-06-28 22:44:09.404700317 +0200
+++ libzip-0.11.2/lib/zip_close.c	2014-06-28 22:49:05.657996641 +0200
@@ -594,20 +594,22 @@ _zip_create_temp_output(struct zip *za,
     char *temp;
     int tfd;
     FILE *tfp;
-    
+
     if (za->tempdir) {
-        if ((temp=(char *)malloc(strlen(za->tempdir)+13)) == NULL) {
+        int len = strlen(za->tempdir) + 13;
+        if ((temp=(char *)malloc(len)) == NULL) {
             _zip_error_set(&za->error, ZIP_ER_MEMORY, 0);
             return NULL;
         }
-        sprintf(temp, "%s/.zip.XXXXXX", za->tempdir);
+        snprintf(temp, len, "%s/.zip.XXXXXX", za->tempdir);
     }
     else {
-        if ((temp=(char *)malloc(strlen(za->zn)+8)) == NULL) {
+        int len = strlen(za->zn) + 8;
+        if ((temp=(char *)malloc(len)) == NULL) {
             _zip_error_set(&za->error, ZIP_ER_MEMORY, 0);
             return NULL;
         }
-        sprintf(temp, "%s.XXXXXX", za->zn);
+        snprintf(temp, len, "%s.XXXXXX", za->zn);
     }
 
     if ((tfd=mkstemp(temp)) == -1) {
--- libzip-0.11.2/lib/zip.h.0002~	2014-06-28 22:44:09.405700304 +0200
+++ libzip-0.11.2/lib/zip.h	2014-06-28 22:49:50.281438771 +0200
@@ -62,7 +62,7 @@ extern "C" {
 #define ZIP_EXCL             2
 #define ZIP_CHECKCONS        4
 #define ZIP_TRUNCATE         8
-
+#define ZIP_OVERWRITE        ZIP_TRUNCATE
 
 /* flags for zip_name_locate, zip_fopen, zip_stat, ... */
 
--- libzip-0.11.2/lib/zip_open.c.0002~	2013-12-19 12:38:36.000000000 +0100
+++ libzip-0.11.2/lib/zip_open.c	2014-06-28 22:51:57.169852448 +0200
@@ -75,7 +75,9 @@ zip_open(const char *fn, int _flags, int
         
     switch (_zip_file_exists(fn, flags, zep)) {
     case -1:
-	return NULL;
+        if (!(flags & ZIP_OVERWRITE))
+	    return NULL;
+	// fallthrough intended
     case 0:
 	return _zip_allocate_new(fn, flags, zep);
     default:
@@ -468,7 +470,7 @@ _zip_file_exists(const char *fn, unsigne
     }
     
     if (stat(fn, &st) != 0) {
-	if (flags & ZIP_CREATE)
+	if (flags & ZIP_CREATE || flags & ZIP_OVERWRITE)
 	    return 0;
 	else {
 	    set_error(zep, NULL, ZIP_ER_OPEN);
