--- libzip-1.3.2/lib/Makefile.am.0000~	2017-11-20 21:55:04.145766254 +0100
+++ libzip-1.3.2/lib/Makefile.am	2017-11-20 21:57:00.747635688 +0100
@@ -5,8 +5,7 @@ AM_CFLAGS=	@CFLAG_VISIBILITY@
 
 lib_LTLIBRARIES = libzip.la
 noinst_HEADERS = zipint.h gladman-fcrypt.h
-include_HEADERS = zip.h
-nodist_include_HEADERS = zipconf.h
+include_HEADERS = zip.h zipconf.h
 
 # also update CMakeLists.txt when changing version
 libzip_la_LDFLAGS=-no-undefined -version-info 5:0:0
